skyportal/models/user_notification.py

```diff
@@ -100,21 +101,29 @@ def user_preferences(target, notification_setting, resource_type):
         return
     if not target.user:
         return
-
-    if not target.user.preferences:
+    if not notification_setting:
```

Comment:
If I understand correctly `notification_setting` is a global boolean that turns all notifications on/off?

Reply:
No! The method `user_preferences(target, notification_setting, resource_type)` takes a target (that should contain a user, text and URL), a notification_setting (not a boolean, but a string that can be "email", "SMS" or "slack") and a `resource_type` (a string like "sources", "group_admission_request", ...).
This method aims to return the user `preferences` IF there are any for the user in the target, for a given `resource_type`, for a given `notification_setting`. In the methods right below it, that send either slack, emails or SMS notification, we simply pass the associated `notification_setting`.
Then, in those methods, if there are no user preferences for that `resource_type` + `notification_setting`, we do not send a notification. If the method returned preferences (meaning that the user activated that notification type), we send the notification accordingly.

Now, I took the opportunity of adding some verification on the `notification_setting`. As the method is only called in 3 places, and the notification_setting is always either "email", "SMS", or "slack", it should not be None, ever. But I thought that adding it would prevent any errors if the user_notification models are modified later on, let's say to add new notification_settings for example.

Reply:
Ok that makes sense. I would maybe change the parameter name to notification_type just to make that clearer.

Reply:
Also I would either make this conditional `if notification_setting is None` to be explicit, or if you really want to also catch things like empty string, at least comment that this is what you are looking to grab. 
I would also consider just doing `if not isinstance(notification_setting, str)` which really does what you want, I think.